core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/TokenUtils.scala

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils._
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import java.time.Instant
+import java.util.UUID
+import org.apache.spark.SparkContext
+import scala.reflect.runtime.currentMirror
+import scala.reflect.runtime.universe._
+import spray.json.DefaultJsonProtocol.{StringJsonFormat, jsonFormat3}
+import spray.json.RootJsonFormat
+
+case class MwcToken (TargetUriHost: String, CapacityObjectId: String, Token: String)
+object TokenUtils {
+  var AADToken: String = ""
+  val MwcWorkloadTypeMl = "ML"
+
+  def getAccessToken: String = {
+    if (isTokenValid(Option(this.AADToken)))
+      this.AADToken
+    else {
+      refreshAccessToken()
+      this.AADToken
+    }
+  }
+
+  def getAccessToken(tokenType: String): String = {
+
+    val objectName = "com.microsoft.azure.trident.tokenlibrary.TokenLibrary"
+    val mirror = currentMirror
+    val module = mirror.staticModule(objectName)
+    val obj = mirror.reflectModule(module).instance
+    val objType = mirror.reflect(obj).symbol.toType
+    val methodName = "getAccessToken"
+    val methodSymbols = objType.decl(TermName(methodName)).asTerm.alternatives
+    val argType = typeOf[String]
+
+    val selectedMethodSymbol = methodSymbols.find { m =>
+      m.asMethod.paramLists match {
+        case List(List(param)) => param.typeSignature =:= argType
+        case _ => false
+      }
+    }.getOrElse(throw new NoSuchMethodException(s"Method $methodName with argument type $argType not found"))
+
+    val methodMirror = mirror.reflect(obj).reflectMethod(selectedMethodSymbol.asMethod)
+    methodMirror(tokenType).asInstanceOf[String]
+  }
+
+  private def isTokenValid(tokenOption: Option[String]): Boolean = {
+    tokenOption match {
+      case Some(token) if token.nonEmpty =>
+        try {
+          val tokenParser = new FabricTokenParser(token)
+          val expiryEpoch = tokenParser.getExpiry
+          val now = Instant.now().getEpochSecond
+          now < expiryEpoch - 60
+        } catch {
+          case e: InvalidJwtTokenException =>
```

Comment:
Don’t catch errors unless you need to

Reply:
I would keep this try catch here because if the cached token is invalid, I would still want the flow to move to next step to refresh the token in TokenUtils::getAccessToken